packaging/linux/amazon-cloudwatch-agent.spec

```diff
@@ -82,7 +82,7 @@ if ! grep "^cwagent:" /etc/group >/dev/null 2>&1; then
 fi
 
 if ! id cwagent >/dev/null 2>&1; then
-    useradd -r -M cwagent -d /home/cwagent -g cwagent -c "Cloudwatch Agent" -s $(test -x /sbin/nologin && echo /sbin/nologin || (test -x /usr/sbin/nologin && echo /usr/sbin/nologin || (test -x /bin/false && echo /bin/false || echo /bin/sh))) >/dev/null 2>&1
```

Comment:
@yimuniao What makes you think it is weird? It is pretty standard for a system account to configure their home dir to be the root of their application dir. It would seem a lot more weird and not standard to be configuring the home dir to something that doesn't exist (Hence the CIS control specifically calling it out).

Some samples from /etc/passwd

```
www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
ntp:x:105:109::/home/ntp:/bin/false
proftpd:x:106:65534::/run/proftpd:/bin/false
postfix:x:109:112::/var/spool/postfix:/bin/false
sshd:x:112:65534::/run/sshd:/usr/sbin/nologin
libvirt-qemu:x:64055:119:Libvirt Qemu,,,:/var/lib/libvirt:/bin/false
```

Reply:
> @yimuniao What makes you think it is weird? It is pretty standard for a system account to configure their home dir to be the root of their application dir. It would seem a lot more weird and not standard to be configuring the home dir to something that doesn't exist (Hence the CIS control specifically calling it out).
> 
> Some samples from /etc/passwd
> 
> ```
> www-data:x:33:33:www-data:/var/www:/usr/sbin/nologin
> proftpd:x:106:65534::/run/proftpd:/bin/false
> postfix:x:109:112::/var/spool/postfix:/bin/false
> sshd:x:112:65534::/run/sshd:/usr/sbin/nologin
> libvirt-qemu:x:64055:119:Libvirt Qemu,,,:/var/lib/libvirt:/bin/false
> ```

I need to do more research. The main concern if there is backward compatible issue if some others are alreadying using /home/cwagent. I will go back here soon.

Reply:
Thanks for the consideration @yimuniao.

I would believe that if someone already has the user created, the package won't try to create a new user, since the useradd command is wrapped in an if statement.

https://github.com/aws/amazon-cloudwatch-agent/blob/2f019fdbc40a8e03901a75f161cabba923f90ca2/packaging/linux/amazon-cloudwatch-agent.spec#L84

That being said, it doesn't seem that the cwagent user utilizes the home dir in any case. It seems to run totally fine in my environment with a non existent home dir.

Reply:
> Thanks for the consideration @yimuniao.
> 
> I would believe that if someone already has the user created, the package won't try to create a new user, since the useradd command is wrapped in an if statement.
> 
> https://github.com/aws/amazon-cloudwatch-agent/blob/2f019fdbc40a8e03901a75f161cabba923f90ca2/packaging/linux/amazon-cloudwatch-agent.spec#L84
> 
> That being said, it doesn't seem that the cwagent user utilizes the home dir in any case. It seems to run totally fine in my environment with a non existent home dir.

What I means is  on old hosts, the home directory is /home/cwagent, on the new hosts, it is other directory, it may cause issue, and the public document update.

Reply:
@yimuniao Any luck in your research?

Reply:
Any luck in your research @yimuniao ?

Reply:
Sorry, I was occupied by other stuff.  I need to discuss with others about this issue after Christmas.

Reply:
Hi @yimuniao want to bump this back up to you. Were you able to review?